### ISSUE TITLE
Show_condition

### ISSUE THREAD

**Issue by nascimento210** [L3MON4D3/LuaSnip#750]
Maybe it's my fault making a bad search in the docs, but I would like to know if there is a show_condition on line begin

**Comment by L3MON4D3** [comment]
Oh, no there isn't.

It's available for `condition` because in that case we know what text of the line corresponds to the trigger, and can incorporate it in the "line-begin"-check.

With `show_condition`, we could implement it such that it only shows a snippet if the cursor is literally at the beginning of the line, but this might remove the completion-item if any completion is hovered (because then the line would not be empty, but include the completion-item-text), even if it is not selected.

I guess this could work if `cmp` is not triggered upon typing any character, but if it is, no idea how that would behave.

Basically, it's a bit hard to define this such that it behaves in an expected way.

**Comment by gennaro-tedesco** [comment]
I could use this issue to follow-up on [this comment](https://github.com/L3MON4D3/LuaSnip/issues/608#issuecomment-1409727585) (so that we avoid creating a new one). My use case is somewhat related to line begin as well. In the below video you can see that I define
```lua
show_condition = function()
  return vim.api.nvim_get_current_line() == ""
end,
```
and nevertheless the snippet is shown whether or not the line is empty.

![Screen-Recording-2023-01-31-at-1](https://user-images.githubusercontent.com/15387611/215737166-fc623137-8fd8-4768-a57d-928231c35acf.gif)

Interestingly enough, if I change the condition to trivially return `true` or `false` then it _does_ behave as intended. Might it be that the method fails to evaluate `vim.api` (or other neovim APIs) in insert mode?

**Comment by L3MON4D3** [comment]
Hmm, yeah that might be it, could you print the output of `current_line`?

**Comment by gennaro-tedesco** [comment]
> could you print the output of current_line?
```lua
show_condition = function()
  print(vim.api.nvim_get_current_line()) <--- printing the current line
  return vim.api.nvim_get_current_line() == ""
end,
```
_does_ print the correct value up to the point I decide to start activating the snippet. It just seems as if the condition is correctly evaluated but has no effect on the snippet.

...might it be I am just putting the `{ show_condition }` in the wrong place within the snippet constructor? :p 

Is the construction as shown in the gif correct (I presume so, otherwise I would have gotten an error)?

**Comment by nascimento210** [comment]
> Oh, no there isn't.
> 
> It's available for `condition` because in that case we know what text of the line corresponds to the trigger, and can incorporate it in the "line-begin"-check.
> 
> With `show_condition`, we could implement it such that it only shows a snippet if the cursor is literally at the beginning of the line, but this might remove the completion-item if any completion is hovered (because then the line would not be empty, but include the completion-item-text), even if it is not selected.
> 
> I guess this could work if `cmp` is not triggered upon typing any character, but if it is, no idea how that would behave.
> 
> Basically, it's a bit hard to define this such that it behaves in an expected way.

thank you for your support
